addons/pinniped/config-controller/controllers/v1_controller_test.go

```diff
@@ -400,11 +401,34 @@ var _ = Describe("Controller", func() {
 
 			It("updates all the addon secrets", func() {
 				for _, c := range clusters {
+					// maybe we can isolate each secret in a separate validation func?
+					// Eventually(verifyV1alpha1SecretUpdateFunc(ctx, c, configMapCopy)).Should(Succeed())
 					Eventually(verifySecretFunc(ctx, c, configMapCopy, true)).Should(Succeed())
 				}
 			})
 		})
 
+		When("the configmap gets deleted", func() {
+			BeforeEach(func() {
+				// gotta create it before we delete it :)
+				createObject(ctx, configMap)
+			})
+			
+			It("deletes all the addon secrets", func() {
+				// delete it
+				deleteObject(ctx, configMap)
+				// verify it is deleted
+				Eventually(func(g Gomega) {
+					err := k8sClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)
+					g.Expect(k8serrors.IsNotFound(err)).To(BeTrue())
```

Comment:
`deleteObject` actually does this verification for you.

Reply:
so i hit a race condition. without this, the test failed consistently.